### ISSUE TITLE
Doesn't extract styles from imported js file

### ISSUE THREAD

**Issue by realgs** [webpack-contrib/mini-css-extract-plugin#756]
TLDR:
mini-css-extract doesn't remove css from js when the styles are situated in an imported .js file. Quick reproduction:
in main.js
```
import '../Content/scss/trends.scss';
```
works and links to the stylesheet correctly (only once)

**BUT** when in main.js you put:
```
import './trends.js';
```
where trends.js contains only:
```
import '../Content/scss/trends.scss';
```
it doesn't work and links the stylesheet twice - from js and from html

**Profound description:**
I encountered a problem with mini-css-extract-plugin not extracting styles from a js file. I'm using webpack 5.30.0 & mini-css-extract-plugin 1.5.0. For the main.scss stylesheet everything works like a charm it loads once - styles are referenced only from html. For the other entries I get styles loaded twice, from the html file & from js (like below the 2nd and the 3rd position- references to styles from js and html, both the same styles). 

![image](https://user-images.githubusercontent.com/5749095/118439183-bf689400-b6e5-11eb-94de-af070c5feb17.png)

**main.js**
```
import '../Content/scss/main.scss';

//(...)

const entries = {
    home: () => import(/* webpackChunkName: "home" */ `~/entries/home`),
    trends: () => import(/* webpackChunkName: "trends" */ `~/entries/trends`),
    others: () => import(/* webpackChunkName: "others" */ `~/entries/others`)
};

//CURRENT_ENTRY depend on which page you're on actually - can be home / trends / others`
entries[CURRENT_ENTRY] ? entries[CURRENT_ENTRY]() : entries.others();
```

**trends.js** (the same for home.js and others.js)
```
import '../../Content/scss/trends.scss';
```

**webpack.config.js**
```
const path = require('path');
const webpack = require('webpack');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const VueLoaderPlugin = require('vue-loader/lib/plugin');
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
const { CleanWebpackPlugin } = require('clean-webpack-plugin');

module.exports = (mode) => ({

name: 'name',
context: path.resolve(__dirname, '../Scripts'),
resolve: {
    extensions: ['.js', '.json', '.html', '.vue'],
    alias: {
        '~': path.resolve(__dirname, '../Scripts'),
        'vue$': 'vue/dist/vue.esm.js',
        '@trackers': path.resolve(__dirname, '../Scripts/trackers')
    },
},
entry: {
    main: './main.js',
},
output: {
    path: path.resolve(__dirname, '../Content/bundle'),
    publicPath: '/Content/bundle/',
    filename: '[name].bundle.js',
    chunkFilename: '[id].js?v=[contenthash]' 
},
plugins: [
    new CleanWebpackPlugin(),
    new webpack.ProgressPlugin(),
    new webpack.DefinePlugin({
        'process.env': {
            'ENTRIES': { home: '~/entries/home.js',
                               trends: '~/entries/trends.js' },
            'NODE_ENV': JSON.stringify(mode)
        }
    }),
    new VueLoaderPlugin(),
    new MiniCssExtractPlugin({
        filename: '[contenthash].[name].bundle.css',
    }),
    new BundleAnalyzerPlugin({
        openAnalyzer: false,
        analyzerMode: 'static'
    })
],
module: {
    rules: [
        {
            test: require.resolve('vue'),
            use: [
                {
                    loader: `expose-loader`,
                    options: 'Vue'
                }
            ]
        },
        {
            test: /\.vue$/,
            loader: 'vue-loader'
        },
        {
            test: /\.(js)$/,
            use: [
                'babel-loader'
            ],
            exclude: /node_modules/
        },
        {
            test: /\.(sa|sc|c)ss$/,
            use: [
                {
                    loader: MiniCssExtractPlugin.loader
                },
                'css-loader',
                'postcss-loader',
                'sass-loader'
            ]
        }
    ]
},
optimization: {
    splitChunks: {
        chunks: 'async'
    }
},
target: 'web'
});
```

**BundleConfig.cs**
```
using System.Web.Optimization;

namespace Project
{
    public class BundleConfig
    {
        public static void RegisterBundles(BundleCollection bundles)
        {
            bundles.Add(new ScriptBundle("~/bundles/js/head").Include(
                "~/Scripts/head.js"
            ));

            bundles.Add(new Bundle("~/bundles/js/scripts").Include(
                "~/Content/bundle/main.bundle.js"
            ));

            bundles.Add(new StyleBundle("~/bundles/css/main").Include(
                "~/Content/bundle/*main.bundle.css"
            ));

            bundles.Add(new StyleBundle("~/bundles/css/home").Include(
                "~/Content/bundle/*home.bundle.css"
            ));

            bundles.Add(new StyleBundle("~/bundles/css/trends").Include(
                "~/Content/bundle/*trends.bundle.css"
            ));
        }
    }
}
```
and in the html files the styles are referenced as:
```
@Styles.Render("~/bundles/css/main")
@Styles.Render("~/bundles/css/home")
@Styles.Render("~/bundles/css/trends")
```

When I modify main.js to contain direct import of one of the chunk css styles for example:
```
import '../Content/scss/trends.scss';
```
it works just fine - the styles are not duplicated. It is obviously not a solution cause I don't want to load all the styles in my main.js - they are separate to create chunks for subpages. 

When the styles are imported in a chunk with the whole trends.js 
```
import(/* webpackChunkName: "trendsChunk" */ `~/trends`);
```
it doesn't work and there is a duplicate styles reference

**Comment by alexander-akait** [comment]
Please create reproducible test repo, bug in `VueLoaderPlugin` or with `sideEffects`

**Comment by alexander-akait** [comment]
Closing due to inactivity. Please test with latest version and feel free to reopen if still regressions. Thanks!
